apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -9,18 +9,32 @@ import java.io.File
 open class GraphqlCompiler {
   private val moshi = Moshi.Builder().build()
 
-  fun write(relativePath: String) {
+  fun write(queryFile: File) {
+    val packageName = queryFile.absolutePath.formatPackageName()
     val irAdapter = moshi.adapter(QueryIntermediateRepresentation::class.java)
-    val ir = irAdapter.fromJson(File(relativePath).readText())
+    val ir = irAdapter.fromJson(queryFile.readText())
     // TODO: Handle multiple or no operations
     val operation = ir.operations.first()
     val typeSpecBuilder = OperationTypeSpecBuilder(operation.operationName, operation.fields)
-    JavaFile.builder("test", typeSpecBuilder.build()).build()
-        .writeTo(OUTPUT_DIRECTORY.fold(File("build"), ::File))
+    JavaFile
+      .builder(packageName, typeSpecBuilder.build())
+      .build()
+      .writeTo(OUTPUT_DIRECTORY.fold(File("build"), ::File))
   }
 
   companion object {
     const val FILE_EXTENSION = "graphql"
     val OUTPUT_DIRECTORY = listOf("generated", "source", "apollo")
   }
+
+  private fun String.formatPackageName(): String {
+    val parts = split(File.separatorChar)
+    val srcFolderIndex = parts.indexOfFirst { it == "src" }
+    val graphqlFolderIndex = parts.indexOfFirst { it == "graphql" }
+    if (graphqlFolderIndex - srcFolderIndex != 2) {
+      throw IllegalArgumentException("Files must be organized like src/main/graphql/...")
+    }
+
+    return parts.subList(graphqlFolderIndex + 1, parts.size).dropLast(1).joinToString(".")
+  }
```

Comment:
This method validates that IR json placed under correct `src/[main|test|etc]/graphql/...` folder and converts the absolutePath to package name by striping first parts of the path. So as example in our tests  `src/test/graphql/com/example` will result `com.example`